web/modules/custom/joinup_core/src/Plugin/Field/FieldFormatter/EntityReferenceCustomLinkFormatter.php

```diff
@@ -0,0 +1,235 @@
+    ]);
+    $summary[] = $this->t('Limit: %limit', [
+      '%limit' => $this->getSetting('limit'),
+    ]);
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+    $entities_to_view = $this->getEntitiesToView($items, $langcode);
+    $limit = $this->getSetting('limit');
+    if ($limit !== -1 && $limit < count($entities_to_view)) {
+      // Due to the fact that delta is not yet implemented in SPARQL, in case
+      // not all terms are shown, sort the terms alphabetically. This will keep
+      // the results predictable and consistent across the website.
+      usort($entities_to_view, function (EntityInterface $entity_a, EntityInterface $entity_b): int {
+        return $entity_a->label() <=> $entity_b->label();
+      });
+      $entities_to_view = array_splice($entities_to_view, 0, (int) $this->getSetting('limit'));
+    }
+
+    foreach ($entities_to_view as $delta => $entity) {
+      $path = $this->getSetting('path');
+      $label = $this->getSetting('label');
+      $query_parameters = $this->getQueryParameters($entity);
+
+      $path = $this->token->replace($path, [$entity->getEntityTypeId() => $entity]);
+      $label = $this->token->replace($label, [$entity->getEntityTypeId() => $entity]);
+
+      $url = Url::fromUri($path, ['query' => $query_parameters]);
+      $elements[$delta] = [
+        '#type' => 'link',
+        '#title' => $label,
+        '#url' => $url,
+        '#options' => $url->getOptions(),
+      ];
+    }
+
+    return $elements;
+  }
+
+  /**
+   * Processes the query parameters and encodes the url parameters.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity in the current delta.
+   *
+   * @return array
+   *   An array of processed query parameters.
+   */
+  protected function getQueryParameters(EntityInterface $entity): array {
+    $setting = $this->getSetting('query_parameters');
+    $query_parameters = explode("\n", $setting);
+    if (empty($query_parameters)) {
+      return [];
+    }
+
+    $return = [];
+    foreach ($query_parameters as $query_parameter) {
+      [$name, $value] = explode('|', $query_parameter, 2);
+      $name = trim($name);
+      $value = rtrim($value, "\r");
+      if (empty($name)) {
+        continue;
+      }
+      $return[$name] = $this->token->replace($value, [$entity->getEntityTypeId() => $entity]);
+    }
+
+    return $return;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function checkAccess(EntityInterface $entity) {
+    return $entity->access('view label', NULL, TRUE);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function isApplicable(FieldDefinitionInterface $field_definition) {
+    // This formatter is only available for taxonomy terms.
+    return $field_definition->getFieldStorageDefinition()->getSetting('target_type') == 'taxonomy_term';
```

Comment:
This is also an evidence that the formatter is very specialized. Or do you have in mind other use cases?

Reply:
The same with the relevant answer above.